src/map/battle.cpp

```diff
@@ -4316,49 +4432,33 @@ static void battle_calc_multi_attack(struct Damage* wd, struct block_list *src,s
  * @param sc: Object's status change information
  * @return atkpercent with cap_value(watk,0,USHRT_MAX)
  */
-static unsigned short battle_get_atkpercent(struct block_list* bl, uint16 skill_id, status_change* sc)
+static unsigned short battle_get_atkpercent(struct block_list& bl, uint16 skill_id, status_change& sc)
 {
-	//These skills are not affected by ATKpercent
-	switch (skill_id) {
-#ifndef RENEWAL
-	// Need to be confirmed for renewal as masteries have been coded to apply to those two in renewal
-	case PA_SHIELDCHAIN:
-	case CR_SHIELDBOOMERANG:
-#endif
-	case AM_ACIDTERROR:
-	case CR_GRANDCROSS:
-	case NPC_GRANDDARKNESS:
-	case MO_INVESTIGATE:
-	case MO_EXTREMITYFIST:
-	case PA_SACRIFICE:
-	case NPC_DRAGONBREATH:
-	case RK_DRAGONBREATH:
-	case RK_DRAGONBREATH_WATER:
+	if (!battle_skill_stacks_masteries_vvs(skill_id, 0))
```

Comment:
Was the old implementation wrong or do we abuse a "wrong named" function here?

Reply:
I updated the function to also handle ATKpercent (see comment of the function).
Skills that don't apply masteries/VVS also won't apply all other things (unless Aegis manually coded it into the skill code as with the refine bonus for Grand Cross), so it's better to channel it all through one function rather than having switches everywhere.
It prevents devs from adding a special skill exception only to one thing they just found instead of making it a general rule.